payment-request/payment-request-ctor-currency-code-checks.https.html

```diff
@@ -0,0 +1,362 @@
+    };
+    const details = {
+      total: defaultTotal,
+      shippingOptions: [shippingOption],
+    };
+    try {
+      new PaymentRequest(defaultMethods, details);
+    } catch (err) {
+      assert_unreached(
+        `Unexpected exception with valid shippingOption currency code "${validCurrency}": ${err.message}.`
+      );
+    }
+    shippingOptions.push(shippingOption);
+  }
+  try {
+    const details = Object.assign({}, defaultDetails, { shippingOptions });
+    new PaymentRequest(defaultMethods, details);
+  } catch (err) {
+    assert_unreached(
+      `Unexpected error with multiple valid shppingOptions: ${err.message}.`
+    );
+  }
+}, "Check and canonicalize valid details.shippingOptions amount.");
+
+test(() => {
+  assert_throws(RANGE_ERROR, smokeTest, "Expected smoke test to throw.");
+  for (const invalidCurrency of invalidCurrencyCodes) {
+    const shippingOption = {
+      id: "test",
+      label: "shipping option",
+      amount: { currency: invalidCurrency, value: "5.00" },
+      selected: true,
+    };
+    const details = {
+      total: defaultTotal,
+      shippingOptions: [shippingOption],
+    };
+    assert_throws(
+      RANGE_ERROR,
+      () => {
+        new PaymentRequest(defaultMethods, details);
+      },
+      `Expected RangeError with invalid shippingOption currency code "${invalidCurrency}".`
+    );
+  }
+}, "Check and canonicalize valid details.shippingOptions amount and rethrow RangeError.");
+
+// Process payment details modifiers:
+test(() => {
+  assert_throws(RANGE_ERROR, smokeTest, "Expected smoke test to throw.");
+  for (const validCurrency of wellFormedCurrencyCodes) {
+    const modifier = {
+      supportedMethods: "https://wpt.fyi",
+      total: {
+        label: "Total due",
+        amount: { currency: validCurrency, value: "68.00" },
+      },
+    };
+    const details = {
+      total: defaultTotal,
+      modifiers: [modifier],
+    };
+    try {
+      new PaymentRequest(defaultMethods, details);
+    } catch (err) {
+      assert_unreached(
+        `Unexpected error with valid modifier currency code "${validCurrency}": ${err.message}`
+      );
+    }
+  }
+}, "Check and canonicalize valid modifiers[n].total amount.");
+
+test(() => {
+  assert_throws(RANGE_ERROR, smokeTest, "Expected smoke test to throw.");
+  for (const invalidCurrency of invalidCurrencyCodes) {
+    const modifier = {
+      supportedMethods: "https://wpt.fyi",
+      total: {
+        label: "Total due",
+        amount: { currency: invalidCurrency, value: "68.00" },
+      },
+    };
+    const details = {
+      total: defaultTotal,
+      modifiers: [modifier],
+    };
+    assert_throws(
+      RANGE_ERROR,
+      () => {
+        new PaymentRequest(defaultMethods, details);
+      },
+      `Expected RangeError with invalid shippingOption currency code "${invalidCurrency}".`
+    );
+  }
+}, "Check and canonicalize valid modifiers[n].total amount and rethrow RangeError.");
+
+test(() => {
```

Comment:
Remove this test duplicate to [above](https://github.com/w3c/web-platform-tests/pull/6670/files#diff-b4079994269456941761c85b811c258fR215).